src/coreclr/jit/objectalloc.cpp

```diff
@@ -400,14 +400,10 @@ bool ObjectAllocator::MorphAllocObjNodes()
                 {
                     allocType = OAT_NEWOBJ;
                 }
-                else if (data->IsHelperCall())
+                else if (data->IsCall() && data->AsCall()->IsHelperCall(comp, CORINFO_HELP_NEWARR_1_VC) &&
```

Comment:
```suggestion
                else if (data->IsHelperCall(comp, CORINFO_HELP_NEWARR_1_VC) &&
```

(note the overload is new with #104815 so you may need to merge from main)

Reply:
Done.